numpy/_globals.py

```diff
@@ -89,3 +92,21 @@ def __repr__(self):
 
 
 _NoValue = _NoValueType()
+
+class CopyMode(enum.IntEnum):
+
+    ALWAYS = 1
+    IF_NEEDED = 0
+    NEVER = 2
+
+    def __bool__(self):
+        # For backwards compatiblity
+        if self == CopyMode.ALWAYS:
+            return True
+
+        if self == CopyMode.IF_NEEDED:
+            return False
```

Comment:
Which specific backwards compatibility scenarios did you have in mind here? I'm struggling to think of how this could be important.

Reply:
As of now, `CopyMode.ALWAYS` is same as `True` (behavior wise) and `CopyMode.IF_NEEDED` is same as `False`(behavior wise). So, I thought to at least allow these two values to be casted to the corresponding existing bool values for ease of use and matching with the current implementation.

Reply:
The scenario would mainly be `__array_function__`.  If we do this, we can forward it e.g. in `np.meshgrid` without breaking early adopters compatibility with downstream.  I first thought `reshape`, that uses `obj.reshape()`. But reshape doesn't have a `copy` argument yet :).  A subclass that gets passed into a function that uses `arr.astype(copy=np.CopyMode.IF_NEEDED)` would also hit it.

The other (small?) advantage is that `np.meshgrid` will work correctly with it (it uses `if copy:`), but I am not sure that matters, since it is easy to fix-up and nothing breaks.

All of that is pretty niche, `astype` users are unlikely to be early adopters to replace `copy=False` and it is likely fine if `__array_function__` users get told that they should update the downstream library.

Reply:
Resolving this for now.